ndb_api/db_helpers.go

```diff
@@ -186,19 +186,23 @@ func convertMapToActionArguments(myMap map[string]string) []ActionArgument {
 func setConfiguredActionArguments(database DatabaseInterface, actionArguments map[string]string) error {
 	errMsgRoot := "Setting configured action arguments failed"
 	if actionArguments == nil {
-		return fmt.Errorf("%s! Action arguments cannot be null.", errMsgRoot)
+		return fmt.Errorf("%s! Action arguments cannot be nil", errMsgRoot)
 	}
 
-	allowedAdditionalArguments, err := util.GetAllowedAdditionalArgumentsForType(database.GetDBInstanceType())
+	allowedAdditionalArguments, err := util.GetAllowedAdditionalArguments(database.IsClone(), database.GetInstanceType())
 	if err != nil {
-		return fmt.Errorf("%s! %s.", errMsgRoot, err.Error())
+		return fmt.Errorf("%s! %s", errMsgRoot, err.Error())
 	}
 
-	for name, value := range database.GetDBInstanceAdditionalArguments() {
-		// Only configure correct actionArguments
+	if len(database.GetAdditionalArguments()) > len(allowedAdditionalArguments) {
```

Comment:
Does the NDB server enforce any such limit on the args length or is this some logic local to operator?

Reply:
This is operator logic. If you are passing action arguments > the allowed additional arguments then you are specifying an incorrect additional argument or you are passing a duplicate.

If you pass an incorrect arg in NDB API you get an `Unexpected argument` error. If you pass a duplicate action argument you also get an error.